lib/Bio/EnsEMBL/DataCheck/Checks/MemberProductionCounts.pm

```diff
@@ -0,0 +1,204 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::MemberProductionCounts;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::Utils::SqlHelper;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'MemberProductionCounts',
+  DESCRIPTION    => 'Checks that the gene_member counts are appropriately populated',
+  GROUPS         => ['compara', 'compara_protein_trees'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['CAFE_gene_family', 'family', 'gene_member', 'gene_member_hom_stats', 'gene_tree_root', 'genome_db']
+};
+
+sub tests {
+  my ($self) = @_;
+  my $dba = $self->dba;
+  my $dbc = $dba->dbc;
+  my $helper = $dbc->sql_helper;
+  my $gdb_adap = $dba->get_GenomeDBAdaptor;
+  
+  fk($dbc, "gene_member_hom_stats", "gene_member_id", "gene_member", "gene_member_id");
+  my $constraint = "tree_type = 'tree' AND ref_root_id IS NULL";
+  fk($dbc, "gene_member_hom_stats", "collection", "gene_tree_root", "clusterset_id", $constraint);
+  
```

Comment:
The first fk constraint is now handled by the generic ForeignKeysCompara datacheck. The second one isn't, so it's for you to decide if it makes sense to have it here, or if all fk checks happen in ForeignKeysCompara, and move it there accordingly.

Reply:
Probably better to have all the `fk` checks together in your `ForeignKeysCompara` DC. The `MemberProductionCounts` is a pretty hefty test as it is, so could do with the reduction I am sure. I can remove them from here, and (to confirm I understand) would you like me to the incorporate the second check into `ForeignKeysCompara`?

Reply:
Yes, that's what I meant.